src/ui/default-prop-types.js

```diff
@@ -173,3 +177,44 @@ export const formItemLayoutShape = PropTypes.shape({
     sm: formItemDimensionShape.isRequired
   }).isRequired
 });
+
+export const importTaskShape = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
+  batchId: PropTypes.string.isRequired,
+  taskType: PropTypes.oneOf(Object.values(TASK_TYPE)),
+  processed: true,
+  attempts: [
+    {
+      startedOn: PropTypes.string,
+      completedOn: PropTypes.string,
+      errors: PropTypes.arrayOf(PropTypes.string).isRequired
+    }
+  ],
+  taskParams: {
+    key: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    slug: PropTypes.string.isRequired,
+    language: PropTypes.string.isRequired,
+    updatedOn: PropTypes.string,
+    importedRevision: PropTypes.string,
+    importableRevision: PropTypes.string.isRequired,
+    importType: PropTypes.oneOf(Object.values(DOCUMENT_IMPORT_TYPE))
+  }
+});
+
+export const importBatchShape = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
+  createdBy: userInDocShape.isRequired,
+  createdOn: PropTypes.string.isRequired,
+  completedOn: PropTypes.string,
+  batchType: PropTypes.oneOf(Object.values(BATCH_TYPE)),
+  batchParams: PropTypes.shape({
+    ...importSourceShape
```

Comment:
You can't destructure this shape as it is a data structure created by the `prop-types` package, not a plain JS object (note that in this file there are variables called `...Props` (JS objects) and `...Shape` (objects created by the library))

Reply:
Interesting, I thought you could because I saw further up the 'languageProps" but I see that is not a shape. Maybe this is why I have the warnings. I will read a bit about these shapes to see what is the benefit of declaring them as a "shape" instead of just props like we have the language.